samples/lulesh/lulesh_flux_spec_v2.0.yaml

```diff
@@ -0,0 +1,70 @@
+description:
+    name: lulesh_sample1
+    description: A sample LULESH study that downloads, builds, and runs a parameter study of varying problem sizes and iterations.
+
+env:
+    variables:
+        OUTPUT_PATH: ./sample_output/lulesh
+
+    labels:
+        outfile: $(SIZE.label).$(ITERATIONS.label).log
+
+    dependencies:
+      git:
+        - name: LULESH
+          path: $(OUTPUT_PATH)
+          url: https://github.com/LLNL/LULESH.git
+
+study:
+    make-lulesh:
```

Comment:
This is an interesting change to the YAML description. It makes it a little less verbose. I do miss the list (`-`) syntax though. It made it easier to make out the individual steps at a glance.

Reply:
It's something I've been going back and forth on. I figured it'd be good to just throw the kitchen sink into the rework and see what stuck. My main concern is with how easy it is to mess up the style with the `-` list notation, but I can see merits in it either way.

Thoughts? @gonsie @kcathey @dinatale2 @tadesautels

Reply:
On that note, I was considering something similar for the dependencies as well.

```
dependencies:
    git:
        LULSEH:
            path: $(OUTPUT_PATH)
            url: https://github.com/LLNL/LULESH.git
```

Reply:
I agree that it might be easy to mess up the list notation if you keep the list notation, but consider that your users are likely used to it. I'd suggest taking some existing specs and converting them to the new format to see if you're missing anything. You can also get feedback from your users once you mock up existing ones.

Reply:
@kcathey -- The depends entry is already a list, so that's handled.

In terms of default values for the batch settings, I've been considering that. However, I have also gotten feedback that the batch settings at the top of the file feel kind of out of place. Like it should all be in the steps -- I'm not remembering 100% what the user said, @jsemler do you happen to recall?

One thing I keep revisiting is just having users specify general values in the specification per step and then having more platform specific information passed in separately (or calculated using cluster information).

I also had another thought on the dependencies section of the specification -- I really don't like that it's three layers deep by the time you specify name, type, then path and other details. As someone who writes code for a living, I feel like it's just stylistically gross and it kind of disrupts the flow of writing my code (or specification in this case) because it becomes harder to track the scope I'm writing in. I had brought this new format to @jsemler but never wrote it down anywhere, it's as follows:

```
dependencies:
    LULSEH:
        type: git
        path: $(OUTPUT_PATH)
        url: https://github.com/LLNL/LULESH.git
```

The new format accomplishes two things primarily: makes the spec less tab heavy and also makes it so the name of the dependencies become the focus.

Reply:
@FrankD412 -- I don't think there was any additional information regarding the batch settings feedback.

I think reducing the levels of indentation for listing dependencies is helpful. One advantage of the current version is it groups dependencies by type. I’m not sure how readable the updated format will be for larger lists of dependencies.

Reply:
IMO, I think the updated format is more readable because it is explicit -- more so in longer lists of dependencies. If you have a list of items long enough you'd not need to scroll anywhere to know what type it was since it's 1:1 mapped directly in the entry.

Reply:
I meant readable from an organizational standpoint of all the dependencies. Though, that might not matter as much. Having the dependency type listed as part of the dependency is helpful. There is redundancy if multiple dependencies have the same type though (not sure how much that matters).